q2_quality_control/decontam_assets/run_decontam.R

```diff
@@ -61,12 +61,17 @@ meta_data_cols <-function(metadata_df, control.col){
       control_vec<-metadata_df[,c(index)]
     }
   }
+  # We need to align the metadata to the asv table
+  mapped_ids <- match(rownames(metadata_df), rownames(asv_df))
+  control_vec <- control_vec[mapped_ids]
+  # drop sample IDs not in the asv table
+  control_vec <- na.omit(control_vec)
```

Comment:
This was the important bit, without it, it's basically a random selection of IDs which are set as the "control" as the table ID ordering has nothing to do with the metadata ID ordering.